File: cpp/Characters/RechargeableUseable.c
// Type: Characters.RechargeableUseable
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Characters/RechargeableUseable.cs
// Prior real C# source: none
// --------------------------------

// Characters.RechargeableUseable$$.ctor
// il2cpp: void Characters_RechargeableUseable___ctor (Characters_RechargeableUseable_o* __this, Characters_BaseCharacter_o* owner, float reduceCooldownAmount, const MethodInfo* method);
// 0x42a9220

void Characters_RechargeableUseable___ctor
               (Characters_RechargeableUseable_o *__this,Characters_BaseCharacter_o *owner,
               float reduceCooldownAmount,MethodInfo *method)

{
  (__this->fields)._lastUseTime = -1000.0;
  System_Object___ctor((Il2CppObject *)__this,(MethodInfo *)0x0);
  (__this->fields)._owner = owner;
  il2cpp_runtime_helper_022b4080(&(__this->fields)._owner,owner);
  (__this->fields).Cooldown = 0.0;
  (__this->fields).UsesLeft = -1;
  (__this->fields).MaxUses = -1;
  (__this->fields).ReduceCooldownAmount = reduceCooldownAmount;
  return;
}


// Characters.RechargeableUseable$$ReduceCooldown
// il2cpp: void Characters_RechargeableUseable__ReduceCooldown (Characters_RechargeableUseable_o* __this, const MethodInfo* method);
// 0x42a9280

void Characters_RechargeableUseable__ReduceCooldown
               (Characters_RechargeableUseable_o *__this,MethodInfo *method)

{
  (__this->fields)._lastUseTime = (__this->fields)._lastUseTime - (__this->fields).ReduceCooldownAmount;
  return;
}

File: csharp_body_deref/Scripts/Characters/RechargeableUseable.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Characters.RechargeableUseable
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Characters/RechargeableUseable.c
// Prior real C# source: none (type is newer than the last public source snapshot)
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace Characters;

[Token(Token = "0x20004E1")]
internal abstract class RechargeableUseable : ExtendedUseable
{
	[Token(Token = "0x40015B1")]
	[FieldOffset(Offset = "0x3C")]
	public float ReduceCooldownAmount;

	[Token(Token = "0x60033B8")]
	[Address(RVA = "0x41A9220", Offset = "0x41A9220", VA = "0x41A9220")]
	public RechargeableUseable(BaseCharacter owner, float reduceCooldownAmount = 0f)
	{
	}

	[Token(Token = "0x60033B9")]
	[Address(RVA = "0x41A9280", Offset = "0x41A9280", VA = "0x41A9280")]
	public void ReduceCooldown()
	{
	}
}
